packages/web-components/fast-foundation/src/select/select.ts

```diff
@@ -0,0 +1,224 @@
+import { Option } from "../option/option";
+import { Listbox } from "../listbox/listbox";
+
+/**
+ * A Select Custom HTML Element.
+ * Implements the {@link https://www.w3.org/TR/wai-aria-1.1/#select | ARIA select }.
+ *
+ * @public
+ */
+export class Select extends FormAssociated<HTMLSelectElement> {
+    protected proxy: HTMLSelectElement = document.createElement("select");
+
+    @attr
+    public autocomplete: string = "off";
+
+    @attr({ attribute: "autofocus", mode: "boolean" })
+    public autofocus: boolean = false;
+
+    /**
+     * The open attribute
+     */
+    @attr({ attribute: "open", mode: "boolean" })
+    public open: boolean;
+    private openChanged(): void {
+        if (this.$fastController.isConnected) {
+            this.button.setAttribute("aria-expanded", `${this.open}`);
+
+            if (this.open) {
+                this.listbox.focusFirstSelectedOption();
+            }
+        }
+    }
+
+    /**
+     * The container for the indicator icon.
+     * @internal
+     */
+    @observable
+    public indicatorContainer: HTMLElement;
+
+    /**
+     * @internal
+     */
+    @observable
+    public defaultSlottedNodes: Node[];
+
+    /**
+     * @internal
+     */
+    @observable
+    public listbox: Listbox;
+
+    /**
+     * @internal
+     */
+    @observable
+    public options: Option[];
+
+    /**
+     * The value displayed on the button
+     * @public
+     */
+    @observable
+    public displayValue: string;
+
+    /**
+     * The button element that displays the selected value and triggers the visibility of the listbox.
+     * @public
+     */
+    @observable
+    public button: HTMLElement;
+
+    /**
+     * The slot that holds the contents of the button
+     * @public
+     */
+    @observable
+    public slottedButtonContainer: Node[];
+
+    /**
+     * Update the value and display value when the listbox emits a change event.
+     * @param e - Event object
+     */
+    private listboxChangeHandler = (e: Event): void => {
+        const captured = e.target as Option;
+        this.value = captured.value;
+    };
+
+    protected valueChanged(previous: string, next: string): void {
+        super.valueChanged(previous, next);
+        DOM.queueUpdate(() => {
+            const selectedOption = this.listbox.selectedOption;
+            this.displayValue = selectedOption.textContent || selectedOption.value;
+        });
+    }
+
+    private updateForm(): void {
+        const value = this.value ? this.value : null;
+        this.setFormValue(value, value);
+    }
+
+    public keydownHandler = (e: KeyboardEvent): void => {
+        super.keypressHandler(e);
+
+        const keyCode = e.key || e.key.charCodeAt(0);
+
+        if (keyCode === "Escape") {
+            this.open = false;
+            DOM.queueUpdate(() => this.button.focus());
+            return;
+        }
+
+        if ([" ", "Enter"].includes(`${keyCode}`)) {
+            e.preventDefault();
+            e.stopPropagation();
```

Comment:
Discussed offline, but we should avoid `stopPropagation()` because it prevents other code from receiving the event, which is often un-intuitive. 

My general rule-of-thumb when the urge to use `stopPropagation()` arises is to make event handlers better at determining if they should act.